PWGLF/Tasks/spectraCharged.cxx

```diff
@@ -221,11 +221,11 @@ bool chargedSpectra::initParticle(const P& particle)
 {
   vars.isChargedPrimary = false;
   auto pdgParticle = pdg->GetParticle(particle.pdgCode());
-  // if (!pdgParticle || pdgParticle->Charge() == 0.) {
-  if (!pdgParticle || TMath::Abs(pdgParticle->Charge()) != 3.) { // This is only a temporary workaround for isPhysicalPrimary and should be replaced by commented line in future
+  if (!pdgParticle || pdgParticle->Charge() == 0.) {
     return false;
   }
-  vars.isChargedPrimary = particle.isPhysicalPrimary();
+  std::list<int> primparticles{11, 13, 211, 321, 2212, 3112, 3222, 3312, 3334};                                                                                           //(Non-exhaustive) list of charged particles that can constitute primaries
+  vars.isChargedPrimary = particle.isPhysicalPrimary() && (std::find(primparticles.begin(), primparticles.end(), TMath::Abs(particle.pdgCode())) != primparticles.end()); // use PID as temporary workaround
```

Comment:
But here you're just checking that the particle PDG is in the list of the particles defined above, right?

Reply:
Exactly. We're interested in the charged primaries and thus have the (pdgParticle->Charge() == 0.) above and the particle.isPhysicalPrimary() here. But since the isPhysicalPrimary is currently not working correctly, I added this workaround so that we exclude particles that cannot be primaries. This second part is supposed to be removed once isPhysicalPrimary is working properly, while the rest will remain here (that's why I changed it back  (pdgParticle->Charge() == 0.). It won't make any difference right now but will once all is working properly).

Reply:
Ok